packages/graphql/src/auth/get-jwt.ts

```diff
@@ -70,31 +70,47 @@ function getJWT(context: Context): any {
             debug("Skipping verifying JWT as noVerify is not set");
 
             result = jsonwebtoken.decode(token);
-        } else if (jwtConfig.jwkEndpoint) {
-            debug("Verifying JWT using OpenID Public Key Endpoint");
+        } else if (jwtConfig.jwksEndpoint) {
+            debug("Verifying JWT using OpenID Public Key Set Endpoint");
 
-            // Create a JWK Client with a rate limit that
-            // limits the number of calls to our JWK endpoint
+            // Creates a JWKS Client with a rate limit that
+            // limits the number of calls to our JWKS endpoint
             client = new JwksClient({
-                jwksUri: jwtConfig.jwkEndpoint,
+                jwksUri: jwtConfig.jwksEndpoint,
                 rateLimit: true,
                 jwksRequestsPerMinute: 10, // Default Value
                 cache: true, // Default Value
                 cacheMaxEntries: 5, // Default value
                 cacheMaxAge: 600000, // Defaults to 10m
             });
 
+            // Verifies the JWKS asynchronously, returns Promise
             /* eslint-disable-next-line no-inner-declarations */
-            function getKey(header, callback) {
-                client.getSigningKey(header.kid, function (err, key) {
-                    const signingKey = key.getPublicKey();
-                    callback(null, signingKey);
-                });
+            async function verifyJWKS() {
+                function getKey(header, callback) {
+                    // Callback that returns the key the corresponding key[kid]
+                    client.getSigningKey(header.kid, (err, key) => {
+                        const signingKey = key?.getPublicKey() || key?.publicKey || key?.rsaPublicKey;
+                        callback(null, signingKey);
+                    });
+                }
+
+                // Returns a Promise with verification result or error
+                return new Promise((resolve, reject) =>
+                    jsonwebtoken.verify(
+                        token,
+                        getKey,
+                        {
+                            algorithms: ["HS256", "RS256"],
+                        },
+                        function (err, decoded) {
+                            return err ? reject(err) : resolve(decoded);
+                        }
+                    )
+                );
             }
```

Comment:
Could this be moved outside of the `getJWT` function as ESLint is suggesting? Even just in the same file is fine, it just makes it a bit clearer.

Reply:
Sure that's fine, moved it outside the function but not allowing it as a file export